verena/hashServer.go

```diff
@@ -1,136 +1,117 @@
 package main
 
 // usage:
-// To run the server (from command line): go run hashServer.go
+// To run the server (from command line): `go run hashServer.go`
 // To interact with the server, open a separate terminal:
-// To get a value from hashServer: curl -X POST localhost:8090/get -H 'Content-Type: application/json' -d '{"uuid":"VALUE"}'
-// To Put a value to hashServer: curl -X POST localhost:8090/put -H 'Content-Type: application/json' -d '{"uuid":"VALUE", "entry":{"hash":"HASH_VALUE", "version":VERSION, "publicKey":"PUBLIC_KEY","hash_old":"HASH_VALUE", "version_old":VERSION, "publicKey_old":"PUBLIC_KEY"}}'
+// To get a value from hashServer: `curl -X POST localhost:8090/get -H 'Content-Type: application/json' -d '{"uuid":"VALUE"}'`
+// To put a value in hashServer:
+//     curl -X POST localhost:8090/put -H 'Content-Type: application/json'
+//     -d '{"uuid":"VALUE", "entry":{"hash":"HASH_VALUE", "version":VERSION, "publicKey":"PUBLIC_KEY"},
+//     "oldEntry":{"hash":"HASH_VALUE", "version":VERSION, "publicKey":"PUBLIC_KEY"}}'
 
 import (
-	"encoding/json"
-	"fmt"
-	"net/http"
+    "encoding/json"
+    "fmt"
+    "net/http"
 )
 
 type Entry struct {
-	Hash          string `json:"hash"`
-	Version       int    `json:"version"`
-	PublicKey     string `json:"publicKey"`
-	Hash_old      string `json:"hash_old"`
-	Version_old   int    `json:"version_old"`
-	PublicKey_old string `json:"publicKey_old"`
```

Comment:
note that these are exactly the same fields as what an `Entry` has, so rather than creating a new struct for this, we should have the request pass in two `Entry` structs